### PR TITLE
Node notes cli

### PR DESCRIPTION
**Pull request by aehrenkranz**
Capture

### CODE REVIEW

File: node-notes-cli/notes.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const desiredFunction = process.argv[2];
+
+if (desiredFunction === 'read') {
+  const entries = await read();
+  for (const entryNumber in entries.notes) {
+    console.log(`${entryNumber}: ${entries.notes[entryNumber]}`);
+  }
+}
+if (desiredFunction === 'create') {
+  create();
```
Comment (dseongdev) [inline node-notes-cli/notes.ts]: these methods should take in a argument

File: node-notes-cli/notes.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+if (desiredFunction === 'update') {
+  update();
+}
+
+async function read(): Promise<Data> {
```
Comment (dseongdev) [inline node-notes-cli/notes.ts:24]: needs argument

File: node-notes-cli/notes.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  try {
+    const resultsJSON = await readFile('data.json', 'utf-8');
+    results = JSON.parse(resultsJSON);
+  } catch (error) {
+    console.log(error);
+  }
+  return results;
+}
+
+async function create(): Promise<void> {
```
Comment (dseongdev) [inline node-notes-cli/notes.ts]: needs argument

File: node-notes-cli/notes.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    data.nextId++;
+    const newData = JSON.stringify(data, null, 2);
+    results = await writeFile('data.json', newData);
+  } catch (error) {
+    console.log(error);
+  }
+  return results;
+}
+
+async function update(): Promise<void> {
```
Comment (dseongdev) [inline node-notes-cli/notes.ts]: needs argument

File: node-notes-cli/notes.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import { writeFile, readFile } from 'node:fs/promises';
+type Data = {
+  nextId: number;
+  notes: Record<string, string>;
+};
+const desiredFunction = process.argv[2];
```
Comment (dseongdev) [inline node-notes-cli/notes.ts:6]: argv at index 2 is not the only argument this app needs

**Review by dseong0224 — CHANGES_REQUESTED** [review]
pls review comments

**Review by dseong0224 — APPROVED** [review]

**Pull request merged**